OpenRA.Game/Map/ActorInitializer.cs

```diff
@@ -53,7 +54,23 @@ public class FacingInit : IActorInit<int>
 		[FieldFromYamlKey] readonly int value = 128;
 		public FacingInit() { }
 		public FacingInit(int init) { value = init; }
-		public int Value(World world) { return value; }
+		public virtual int Value(World world) { return value; }
+	}
+
+	public class DynamicFacingInit : FacingInit
+	{
+		readonly Func<int> func;
+		public DynamicFacingInit(Func<int> func) { this.func = func; }
+		public override int Value(World world) { return func(); }
+	}
+
+	public class BodyAnimationFrameInit : IActorInit<uint>
```

Comment:
It looks like all of these `*Init`s except `OwnerInit` can be moved into mod code.  Can you create an `ActorInitializers.cs` file in Mods.Common for them?